File: applicationKT/app/src/main/java/com/englishapp/data/model/Token.kt
package com.englishapp.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Token(
    @Json(name = "id") val id: Long,
    @Json(name = "token_type_id") val tokenTypeId: Long,
    @Json(name = "owner_id") val ownerId: Long? = null,
    @Json(name = "code") val code: String,
    @Json(name = "remaining_uses") val remainingUses: Int? = null,
    @Json(name = "expires_at") val expiresAt: String? = null,
    @Json(name = "status") val status: String = "active",
    @Json(name = "token_type") val tokenType: TokenType? = null,
    @Json(name = "owner") val owner: User? = null,
    @Json(name = "created_at") val createdAt: String? = null,
    @Json(name = "updated_at") val updatedAt: String? = null
)

File: app/Models/Token.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Token extends Model
{
    protected $fillable = [
        'token_type_id',
        'owner_id',
        'code',
        'remaining_uses',
        'expires_at',
        'status',
    ];

    protected $casts = [
        'expires_at' => 'datetime',
    ];

    public function tokenType()
    {
        return $this->belongsTo(TokenType::class);
    }

    public function owner()
    {
        return $this->belongsTo(User::class, 'owner_id');
    }

    public function enrollments()
    {
        return $this->hasMany(Enrollment::class);
    }

    public function transfers()
    {
        return $this->hasMany(TokenTransfer::class);
    }
}
